stdlib/fun.mli

```diff
@@ -61,3 +63,67 @@ exception Finally_raised of exn
     an unexpected exception or a programming error. As a general rule,
     one should not catch a [Finally_raised] exception except as part of
     a catch-all handler. *)
+
+(** {1:examples Examples}
+
+    {2 Combinators}
+
+    One common mode of usage for {!combinators} is to create and pass anonymous
+    functions as arguments to higher-order functions, without needing to specify
+    [(fun ... -> ...)] explicitly.
+
+    The examples below will be demonstrating this mainly with the {!module:List}
+    module.
+
+    {!val:id}
+    {[
+    # List.init 3 Fun.id;;
+    - : int list = [0; 1; 2]
+
+    # List.find_all Fun.id [false; false; true; false];;
+    - : bool list = [true]
+    ]}
+
+    {!val:const}
+    {[
+    # List.init 3 (Fun.const 0)
+    - : int list = [0; 0; 0]
+    ]}
+    One difference between applying [const (...)] and passing the lambda
+    [(fun _ -> ...)] directly is that the expression denoted [(...)] will be
+    evaluated once, and the function [const] returns will only have access to
+    the result of evaluating [(...)]. So really, [const (...)] is not equivalent
+    to [(fun _ -> ...)], but rather to [let result = ... in (fun _ -> result)].
```

Comment:
The fact that `const` evalutate its argument once is a remark. I would advise to describe first what  `const` is before describing finer use points. 

Moreover, the paragraph feels too long and verbose. I would argue mentioning that `const` evaluates its argument once while using the (nice) `Random.bool`  example to illustrate the point should be enough.

Reply:
I don't want to repeat what's written in the `const` doc comment which I feel does describe what `const` is.. What do you have in mind? Provide an example implementation `let const x _ = x` maybe?

I'm also worried the bool example on its own wouldn't be too apparent if one doesn't have a strong mental model of how things evaluate already.. I want to at least keep the equivalence fact so that people can experiment with it.

Reply:
You are right that re-explaining `const` would be verbose. Part of the issue is maybe trying to explain `const` out of context. If we apply the idea to have a thematic `building a Seq.t`, there is probably no need to do so.

My fear is that the equivalence explanation increases the probability that readers will not read the paragraph nor the example. We should keep in mind that the example should target the audience that skipped the function definition.

I would propose at least to start with a short sentence: `beware "const" evaluates its argument once`, then the `Random` example and its explanation. I am not sure if we need to write down the equivalence.